File: software_challenge_sdk/src/framework/sc/shared/PlayerColor.java
package sc.shared;

public enum PlayerColor {

  RED, BLUE;

  /**
   * liefert die Spielerfarbe des Gegners dieses Spielers
   * @return Spielerfarbe des Gegners
   */
  public PlayerColor opponent() {
    return this == RED ? BLUE : RED;
  }

}

File: socha-sdk/src/framework/sc/shared/PlayerColor.kt
package sc.shared

enum class PlayerColor(val index: Int, val displayName: String) {
    
    RED(0, "Rot"), BLUE(1, "Blau");
    
    val letter = name.first()
    
    /** Die Spielerfarbe des Gegners dieses Spielers */
    fun opponent(): PlayerColor =
            when (this) {
                RED -> BLUE
                BLUE -> RED
            }
    
}
